### ISSUE TITLE
Breaking blocks causes massive lag (low tps < 10)

### ISSUE THREAD

**Issue by bbruno5** [FabioZumbi12/RedProtect#226]
Tested versions:
RedProtect 7.2.2 b435
RedProtect 7.3.0 b55 Universal

When breaking blocks like dirt, sand, stone,ice, etc. using a tool with efficience 3 or more, like pickaxe and shovel, the server tps slow down massively, from 20 to 10 or less. Disabling the redprotect, the problem don't occurs. Running the spigot tool /timings, I got the confirmation about this plugin, as you can see on the images below from each version.

![deepinscreenshot_select-area_20171126125354](https://user-images.githubusercontent.com/23748196/33241326-e79e5dce-d2aa-11e7-91a6-ea27b42be070.png)

![deepinscreenshot_select-area_20171126132339](https://user-images.githubusercontent.com/23748196/33241486-2f0edef2-d2ad-11e7-9721-e25a694fa8e8.png)

**Comment by Red3Tango** [comment]
What version of MC and Spigot are you running?
Curious as I was unable to reproduce on my server, was insta-mining a few thousand dirt blocks so turned on timings. I had less than 0.1%, and avg 0.04ms on RedProtect, TPS at 19.92. I am also using RP-b55.

**Comment by bbruno5** [comment]
Spigot 1.11.2

Client minecraft on any version, tested from 1.8 to 1.12.2

**Comment by FabioZumbi12** [comment]
I cant solve tps issues since can be caused by any factor/server config/server hardware. This changes between servers/users.

**Comment by bbruno5** [comment]
So, after exaustive tests I found the problem. With and without other plugins, I got low tps, but when I reset the redprotect settings, the lag disappeared. And so, testing one by one, I found that the problem are on mysql connection. I think that 60, maybe more event calls (BlockBreakEvent) per second, calling for mysql to verify if the player is in a RP is causing the lag issues, and it makes sents (due to massive mysql calls per second). Disabling mysql data and using yml files, the lag disappeared 100%.
